x/cdp/keeper/draw.go

```diff
@@ -144,7 +143,10 @@ func (k Keeper) RepayPrincipal(ctx sdk.Context, owner sdk.AccAddress, collateral
 	cdp.AccumulatedFees = cdp.AccumulatedFees.Sub(feePayment)
 
 	// decrement the total principal for the input collateral type
-	k.DecrementTotalPrincipal(ctx, cdp.Type, feePayment.Add(principalPayment))
+	k.DecrementTotalPrincipal(ctx, cdp.Type, totalRepayment)
+
+	// increment total fees collected
+	k.IncrementFeesCollected(ctx, cdp.Principal.Denom, feePayment.Amount)
```

Comment:
I don't think this approach to calculating 'fees collected' is correct. The way the code is written, it is definitionally true that feesAssessed = feesRepaid. Either the CDP owner will call `RepayPrincipal`, which always collects fees before principal, or the CDP will get liquidated, which attempts to raise funds equal to `fees + principal`. Calling `IncrementFeesCollected` in `RepayPrincipal` doesn't tell us how much fees were collected in total, just how much was repaid this way vs getting liquidated. 

I think the best approach is to call `IncrementFeesCollected` in `UpdateFeesForAllCdps` and perhaps change the function to `IncrementFeesAssessed`

Reply:
That makes sense, I'll make those changes.

Reply:
However, I do think it's valuable to be able to differentiate between outstanding fees and past fees paid (protocol profit) and this approach doesn't handle that differentiation.